engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,609 @@
+   * normalization. See {@code Text_Utils.compare_to}.
+   */
+  @TruffleBoundary
+  @Specialization(guards = {
+      "selfInterop.isString(selfString)",
+      "otherInterop.isString(otherString)"
+  }, limit = "3")
+  boolean equalsStrings(Object selfString, Object otherString,
+      @CachedLibrary("selfString") InteropLibrary selfInterop,
+      @CachedLibrary("otherString") InteropLibrary otherInterop) {
+    String selfJavaString;
+    String otherJavaString;
+    try {
+      selfJavaString = selfInterop.asString(selfString);
+      otherJavaString = otherInterop.asString(otherString);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return Normalizer.compare(
+        selfJavaString,
+        otherJavaString,
+        Normalizer.FOLD_CASE_DEFAULT
+    ) == 0;
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isExecutable(selfFunction)",
+      "otherInterop.isExecutable(otherFunction)",
+      "selfInterop.isMemberInvocable(selfFunction, EQUALS_MEMBER_NAME)",
+      "otherInterop.isMemberInvocable(otherFunction, EQUALS_MEMBER_NAME)",
+  }, limit = "3")
+  boolean equalsFunctions(Object selfFunction, Object otherFunction,
+      @CachedLibrary("selfFunction") InteropLibrary selfInterop,
+      @CachedLibrary("otherFunction") InteropLibrary otherInterop,
+      @CachedLibrary(limit = "3") InteropLibrary retValueInterop) {
+    Object ret;
+    try {
+      ret = selfInterop.invokeMember(selfFunction, MethodNames.Function.EQUALS, otherFunction);
+      return retValueInterop.asBoolean(ret);
+    } catch (UnsupportedMessageException | ArityException | UnknownIdentifierException |
+             UnsupportedTypeException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.hasArrayElements(selfArray)",
+      "otherInterop.hasArrayElements(otherArray)"
+  }, limit = "3")
+  boolean equalsArrays(Object selfArray, Object otherArray,
+      @CachedLibrary("selfArray") InteropLibrary selfInterop,
+      @CachedLibrary("otherArray") InteropLibrary otherInterop,
+      @Cached EqualsAnyNode equalsNode
+      ) {
+    try {
+      long selfSize = selfInterop.getArraySize(selfArray);
+      if (selfSize != otherInterop.getArraySize(otherArray)) {
+        return false;
+      }
+      for (long i = 0; i < selfSize; i++) {
+        Object selfElem = selfInterop.readArrayElement(selfArray, i);
+        Object otherElem = otherInterop.readArrayElement(otherArray, i);
+        if (!equalsNode.execute(selfElem, otherElem)) {
+          return false;
+        }
+      }
+      return true;
+    } catch (UnsupportedMessageException | InvalidArrayIndexException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Behind truffle boundary, because comparing exceptions is not expected to be on fast path.
+   * @return True if the givne exceptions are the same
```

Comment:
```suggestion
```

Not sure if this comment is needed really. Other specializations do not have such comments (and it doesn't really add much), so let's be consistent.